src/File.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace React\Http;
+
+use React\Stream\ReadableStreamInterface;
+
+class File implements FileInterface
+{
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var string
+     */
+    protected $filename;
+
+    /**
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * @var ReadableStreamInterface
+     */
+    protected $stream;
+
+    /**
+     * @param string $name
+     * @param string $filename
+     * @param string $type
+     * @param ReadableStreamInterface $stream
+     */
+    public function __construct($name, $filename, $type, ReadableStreamInterface $stream)
+    {
+        $this->name = $name;
+        $this->filename = $filename;
+        $this->type = $type;
+        $this->stream = $stream;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
```

Comment:
By the way, even though we are working with streams, at the time we've parsed one file and move to the next, we do actually know file size (honestly haven't read the whole implementation here why size is not included) and it will not be changed afterwards, so can be safely stored in object alongside with other info.